prisma/migrations/20241217163642_user_values_improve/migration.sql

```diff
@@ -0,0 +1,90 @@
+                    ROW_NUMBER()  OVER (PARTITION BY "parentId" IS NULL ORDER BY ("weightedVotes"-"weightedDownVotes") desc) AS rank
+                FROM
+                    "Item"
+                WHERE date_trunc(date_part, created_at AT TIME ZONE 'UTC' AT TIME ZONE 'America/Chicago') = period.t
+                AND "weightedVotes" > 0
+                AND "deletedAt" IS NULL
+                AND NOT bio
+                AND ("invoiceActionState" IS NULL OR "invoiceActionState" = 'PAID')
+            ) x
+            WHERE x.percentile <= percentile_cutoff
+        ),
+        -- get top upvoters of top posts and comments
+        upvoter_islands AS (
+            SELECT "ItemAct"."userId", item_ratios.id, item_ratios.ratio, item_ratios."parentId",
+                "ItemAct".msats as tipped, "ItemAct".created_at as acted_at,
+                ROW_NUMBER() OVER (partition by item_ratios.id order by "ItemAct".created_at asc)
+                - ROW_NUMBER() OVER (partition by item_ratios.id, "ItemAct"."userId" order by "ItemAct".created_at asc) AS island
+            FROM item_ratios
+            JOIN "ItemAct" on "ItemAct"."itemId" = item_ratios.id
+            WHERE act = 'TIP'
+            AND date_trunc(date_part, "ItemAct".created_at AT TIME ZONE 'UTC' AT TIME ZONE 'America/Chicago') = period.t
+            AND ("ItemAct"."invoiceActionState" IS NULL OR "ItemAct"."invoiceActionState" = 'PAID')
+        ),
+        -- isolate contiguous upzaps from the same user on the same item so that when we take the log
+        -- of the upzaps it accounts for successive zaps and does not disproportionately reward them
+        -- quad root of the total tipped
+        upvoters AS (
+            SELECT "userId", upvoter_islands.id, ratio, "parentId", GREATEST(power(sum(tipped) / 1000, 0.25), 0) as tipped, min(acted_at) as acted_at
+            FROM upvoter_islands
+            GROUP BY "userId", upvoter_islands.id, ratio, "parentId", island
+        ),
+        -- the relative contribution of each upvoter to the post/comment
+        -- early component: 1/ln(early_rank + e - 1)
+        -- tipped component: how much they tipped relative to the total tipped for the item
+        -- multiplied by the relative rank of the item to the total items
+        -- multiplied by the trust of the user
+        upvoter_ratios AS (
+            SELECT "userId", sum((early_multiplier+tipped_ratio)*ratio*CASE WHEN users.id = ANY (handicap_ids) THEN handicap_zap_mult ELSE users.trust+0.1 END) as upvoter_ratio,
+                "parentId" IS NULL as "isPost", CASE WHEN "parentId" IS NULL THEN 'TIP_POST' ELSE 'TIP_COMMENT' END as type
+            FROM (
+                SELECT *,
+                    1.0/LN(ROW_NUMBER() OVER (partition by upvoters.id order by acted_at asc) + EXP(1.0) - 1) AS early_multiplier,
+                    tipped::float/(sum(tipped) OVER (partition by upvoters.id)) tipped_ratio
+                FROM upvoters
+                WHERE tipped > 2
```

Comment:
why was this changed from `> 0` to `> 2`?

Reply:
This enforces a minimum zap to be considered for zapping rewards. Because we use the quad root of the zap amount, this means the minimum zap for consideration is `2 = x^0.25` and `x = 16 sats`. 

It's arbitrary.

Reply:
Maybe I should increase it more.

Reply:
> Maybe I should increase it more.

Sounds good, I would rather make it too large than too small

> An alternative to setting this limit would to weight zap order by the zap amount.

This would mean that zap amount now becomes more important than zap order, no?